server/src/api/post.rs

```diff
@@ -162,6 +163,13 @@ impl Perform for Oper<CreatePost> {
       return Err(APIError::err("site_ban").into());
     }
 
+    if data.url.is_some() {
+      match Url::parse(data.url.as_ref().unwrap()) {
```

Comment:
Instead of is_some unwrap, do an `if let` here if you would.

```
    if let Some(url) = data.url.as_ref() {
      match Url::parse(url) {
        Ok(_t) => (),
        Err(_e) => return Err(APIError::err("invalid_url").into()),
      }
    }
```

Reply:
Oh also and add "invalid_url" to `en.json`

Reply:
done.